wire/common.go

```diff
@@ -390,31 +508,30 @@ func ReadVarInt(r io.Reader, pver uint32) (uint64, error) {
 // on its value.
 func WriteVarInt(w io.Writer, pver uint32, val uint64) error {
 	if val < 0xfd {
-		_, err := w.Write([]byte{uint8(val)})
-		return err
+		return binarySerializer.PutUint8(w, uint8(val))
 	}
 
 	if val <= math.MaxUint16 {
-		var buf [3]byte
-		buf[0] = 0xfd
-		binary.LittleEndian.PutUint16(buf[1:], uint16(val))
-		_, err := w.Write(buf[:])
-		return err
+		err := binarySerializer.PutUint8(w, 0xfd)
+		if err != nil {
+			return err
+		}
+		return binarySerializer.PutUint16(w, littleEndian, uint16(val))
 	}
 
 	if val <= math.MaxUint32 {
-		var buf [5]byte
-		buf[0] = 0xfe
-		binary.LittleEndian.PutUint32(buf[1:], uint32(val))
-		_, err := w.Write(buf[:])
-		return err
+		err := binarySerializer.PutUint8(w, 0xfe)
+		if err != nil {
+			return err
+		}
+		return binarySerializer.PutUint32(w, littleEndian, uint32(val))
 	}
 
-	var buf [9]byte
-	buf[0] = 0xff
-	binary.LittleEndian.PutUint64(buf[1:], val)
-	_, err := w.Write(buf[:])
-	return err
+	err := binarySerializer.PutUint8(w, 0xff)
+	if err != nil {
+		return err
```

Comment:
This may just be Github's formatting (I'm viewing this using the split diff), but it looks like this line hasn't gone through `go fmt`.